.github/workflows/ci.yml

```
@@ -0,0 +1,48 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+
+  check:
+    needs: []
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Set up GitHub Actions"
+      uses: actions/checkout@v2
+    - name: "Set up Python 3.6"
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+    - name: "Install Python dependencies"
+      run: |
+        pip install ".[dev]"
+    - name: "Check code format"
+      run: |
+        make check
+
+  test:
+    needs: [check]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+    steps:
+    - name: "Set up GitHub Actions"
+      uses: actions/checkout@v2
+    - name: "Set up Python ${{ matrix.python-version }}"
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: "Install Python dependencies"
+      run: |
+        pip install ".[dev]"
```

```suggestion
        python -m pip install --upgrade pip setuptools wheel
        python -m pip install --ignore-installed --quiet --no-cache-dir .[dev]
        python -m pip list
```

I usually like to upgrade `pip`, `setuptools`, and `wheel` so that the CI is testing what a brand new user would probably get and then in addition to a few small things `pip list` at the end so that we can easily see in the logs between runs if there are dependency version changes.

---

I am fine with this change.

Although, please consider that installing `setup.py` particular set of dependencies (in this case`.[dev]`) without quotes (`".[dev]"`), does not work on certain shells (i.e. [zsh](https://ohmyz.sh/)).

---

sure, but the CI machines don't run `zsh`. This is good to know though, as I didn't know `zsh` was that picky. :(